docs/developer_documentation/ref-permissions-api.md

```diff
@@ -0,0 +1,574 @@
+- Users inherit permissions from their roles, and roles from their parentroles.
+- The access control list for a resource can have a parent, in that case permissions on the parent also grant permissions on the child ACL's. In the current MOLGENIS system this is the case for entity types and packages, both inherit permissions from the package in which they reside..
+
+## Managing row level security
+### Getting all resource types in the system
+##### Endpoint
+```
+GET https://molgenis.mydomain.example/api/permissions/v1/types
+```
+##### Parameters
+URL: none
+Query: none
+
+##### Response
+List of ACL types available in the system.
+##### Example 
+Response:
+```
+[
+       "package",
+       "entity-sys_job_ResourceDownloadJobExecution",
+       "entity-sys_FileMeta",
+       "entity-sys_ImportRun",
+       "entity-sys_job_OneClickImportJobExecution",
+       "entity-sys_job_ResourceDeleteJobExecution",
+       "entity-sys_job_ResourceCopyJobExecution",
+       "entityType",
+       "plugin",
+       "entity-hospital_cardiology_patients"
+   ]
+   ```
+
+### Getting all suitable permissions for a resource type
+##### Endpoint
+```
+GET https://molgenis.mydomain.example/api/permissions/v1/types/{typeId}")
+```
+##### Parameters
+URL: TypeId as described in the [parameters section](##Parameters)
+
+##### Response
+A list of permissions that can be used for this resource type.
+
+##### Example 
+Request:
+```https://molgenis.mydomain.example/api/permissions/v1/types/permissions/entityType```
+Response:
+```[
+       "READMETA",
+       "READ",
+       "COUNT",
+       "WRITEMETA",
+       "WRITE"
+   ]
+   ```
+
+### Creating a new resource type in the system (Row level securing an entity)
+
+Warning: this will not create ACL's for already existing resources of this type.
+You can use the endpoint in the next section 'Creating a new access control list for a resource' to create ACL's for these resources.
+
+##### Endpoint
+```
+POST https://molgenis.mydomain.example/api/permissions/v1/types/{typeId}")
+```
+########## Parameters
+URL: 'typeId' as described in the [parameters section](##Parameters)
+
+########## Response
+201 CREATED
+
+########## Example
+Enable row level security on entity 'hospital_neurology_patients'.
+
+Request:
+```https://molgenis.mydomain.example/api/permissions/v1/types/entity-hospital_neurology_patients```
+
+
+### Creating a new access control list for a resource
+##### Endpoint
+```
+POST https://molgenis.mydomain.example/api/permissions/v1/objects/{typeId}/{objectId}")
+```
+##### Parameters
+URL: 
+- 'typeId' as described in the [parameters section](##Parameters)
+- 'identifier' as described in the [parameters section](##Parameters)
+
+##### Response
+201 CREATED
+
+##### Example
+Add an ACL for 'Patient1' in entity 'hospital_neurology_patients'.
+
+Request:
+```POST https://molgenis.mydomain.example/api/permissions/v1/types/entity-hospital_neurology_patients/Patient1```
+
+### Getting all acls for a class
```

Comment:
This is the first time the term `class` is used. Is it used interchangeably with `typeId`?

Reply:
If it should be called something else than `class`, I think the following error message should also be updated:

```
"message": "Acl class 'entity-it_emx_autoid_testAutoId' already exists.",
"code": "PRM14"
```

Reply:
I'll take a look, this is an artefact of the renaming. It is caused by the fact that both terms are used in spring security as well...